packages/EasyUtils/src/CreditCard/CreditCardNumberValidator.php

```diff
@@ -0,0 +1,148 @@
+<?php
+
+declare(strict_types=1);
+
+namespace EonX\EasyUtils\CreditCard;
+
+final class CreditCardNumberValidator implements CreditCardNumberValidatorInterface
+{
+    private const CARD_TYPES = [
+        // Debit cards must come first, since they have more specific patterns than their credit-card equivalents.
+        'visaelectron' => [
+            'pattern' => '/^4(026|17500|405|508|844|91[37])/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+        'maestro' => [
+            'pattern' => '/^(5(018|0[23]|[68])|6(39|7))/',
+            'length' => [12, 13, 14, 15, 16, 17, 18, 19],
+            'luhn' => true,
+        ],
+        'forbrugsforeningen' => [
+            'pattern' => '/^600/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+        'dankort' => [
+            'pattern' => '/^5019/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+        // Credit cards
+        'visa' => [
+            'pattern' => '/^4/',
+            'length' => [13, 16],
+            'luhn' => true,
+        ],
+        'mastercard' => [
+            'pattern' => '/^(5[0-5]|2[2-7])/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+        'amex' => [
+            'pattern' => '/^3[47]/',
+            'format' => '/(\d{1,4})(\d{1,6})?(\d{1,5})?/',
+            'length' => [15],
+            'luhn' => true,
+        ],
+        'dinersclub' => [
+            'pattern' => '/^3[0689]/',
+            'length' => [14],
+            'luhn' => true,
+        ],
+        'discover' => [
+            'pattern' => '/^6([045]|22)/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+        'unionpay' => [
+            'pattern' => '/^(62|88)/',
+            'length' => [16, 17, 18, 19],
+            'luhn' => false,
+        ],
+        'jcb' => [
+            'pattern' => '/^35/',
+            'length' => [16],
+            'luhn' => true,
+        ],
+    ];
+
+    public function isCreditCardNumberValid(string $number): bool
+    {
+        // Strip non-numeric characters
+        $number = preg_replace('/\D/', '', $number);
```

Comment:
Please fix `preg_replace` -> `\preg_replace`. The same goes for other built-in functions usage in this PR.

Reply:
Fixed